extensions/todoist/src/hooks/useFilterData.ts

```diff
@@ -0,0 +1,28 @@
+import { usePromise, useCachedState } from "@raycast/utils";
+import { useEffect } from "react";
+
+import { Task, getFilterTasks } from "../api";
+
+function useCachedFiltered() {
+  return useCachedState<Task[]>("filter_tasks");
+}
+
+export default function useFilterTasks(filter = "") {
+  // ToDo: Only fetch if sync data has change. I didn't find how to do it since usePromise and useMemo/useEffect can't easily be nested?
+  const { data: filterData, ...rest } = usePromise(async () => {
+    if (filter !== "") {
+      console.log("fetching filter");
+      const data = await getFilterTasks(filter);
+      return data as Task[];
+    }
+  });
+  const [cachedData, setCachedData] = useCachedFiltered();
+
+  useEffect(() => {
+    if (filterData) {
+      setCachedData(filterData);
+    }
+  }, [filterData]);
+
+  return { data: cachedData, ...rest };
+}
```

Comment:
I think it's fine to make a call even when sync data doesn't change. However, you should probably use `useCachedPromise`:

```suggestion
export default function useFilterTasks(filter = "") {
  return useCachedPromise(async () => {
    if (!filter) return []
    
    const data = await getFilterTasks(filter);
    return data as Task[];
  });
}
```